src/profile_manager/forms.py

```diff
@@ -28,8 +34,60 @@ def clean_alias(self):
 
     def save(self, *args, **kwargs):
         super(ProfileForm, self).save(*args, **kwargs)
+
         user = self.instance.user
         user.email = self.cleaned_data['email']
         user.save()
 
         return self.instance
+
+
+class ProfileFormStaffOnly(ProfileForm):
+
+    username = forms.CharField(required=True)
+    first_name = forms.CharField(required=False)
+    last_name = forms.CharField(required=False)
+    staff = forms.BooleanField(required=False)
+    superuser = forms.BooleanField(required=False)
+    active = forms.BooleanField(required=False)
+
+    class Meta(ProfileForm.Meta):
+        fields = ProfileForm._meta.fields + [
+            'username', 'first_name', 'last_name', 'staff', 'superuser', 'active', 'is_TA',
+        ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields['username'].initial = self.instance.user.username
+        self.fields['username'].help_text = "Unique usernames only!"
+        self.fields['first_name'].initial = self.instance.user.first_name
+        self.fields['last_name'].initial = self.instance.user.last_name
+        self.fields['staff'].initial = self.instance.user.is_staff
+        self.fields['superuser'].initial = self.instance.user.is_superuser
+        self.fields['superuser'].help_text = "Superusers can access the admin section of the deck"
+        self.fields['active'].initial = self.instance.user.is_active
+
+    def clean_username(self):
+        username = self.cleaned_data['username']
+
+        if not username:
+            raise ValidationError("Username CANNOT be empty!")
+        elif get_user_model().objects.exclude(username=self.instance.user.username).filter(username=username).exists():
```

Comment:
since im making my own form to change the username i dont have access to the username validators that other user forms have. So its just there to check if the username is unique, and it excludes the user's name from the check to allow the username to remain the same if they dont want to change it

Reply:
So I would refactor and generate a ModelForm with model=user, and add any custom code to your ModelForm (same as already doing with the Profile's ModelForm